Foundation/NSJSONSerialization.swift

```diff
@@ -322,6 +326,93 @@ private struct JSONWriter {
         }
     }
 
+    private func serializeUInt(value: UInt) throws {
+        if value == 0 {
+            writer("0")
+            return
+        }
+        var array: [UInt] = []
+        var stringResult = ""
+        //Maximum length of an UInt
+        array.reserveCapacity(20)
```

Comment:
I've also made these changes, you can find them in the new commit. 
Is this an acceptable way of formalizing it @parkera ?